allauth/account/adapter.py

```diff
@@ -229,14 +229,16 @@ def clean_username(self, username):
         if username.lower() in username_blacklist_lower:
             raise forms.ValidationError(_("Username can not be used. "
                                           "Please use other username."))
-        username_field = app_settings.USER_MODEL_USERNAME_FIELD
-        assert username_field
-        user_model = get_user_model()
-        try:
-            query = {username_field + '__iexact': username}
-            user_model.objects.get(**query)
-        except user_model.DoesNotExist:
-            return username
+        # Skipping database lookups when shallow is True, needed for unique username generation.
+        if not shallow:
+            username_field = app_settings.USER_MODEL_USERNAME_FIELD
+            assert username_field
+            user_model = get_user_model()
+            try:
+                query = {username_field + '__iexact': username}
+                user_model.objects.get(**query)
+            except user_model.DoesNotExist:
+                return username
         raise forms.ValidationError(_("This username is already taken. Please "
```

Comment:
If `shallow` is `True` we always raise a `ValidationError` here, that should not be the case.